source/UOXJSMethods.cpp

```diff
@@ -3597,11 +3598,12 @@ JSBool CChar_ResourceCount( JSContext *cx, JSObject *obj, uintN argc, jsval *arg
 
 	UI16 realId = static_cast<UI16>( JSVAL_TO_INT( argv[0] ));
 	SI32 itemColour = 0;
-	UI32 moreVal = 0;
+	SI64 moreVal = -1;
```

Comment:
Is it worth either declaring this as `SERIAL` or bring it back to UI32, and rather than -1 use INVALIDSERIAL?

Reply:
Possibly, though... still need to handle the possibility of a -1 value being provided as a parameter from the script function, since the syntax supports an optional sectionId param at the end, and when using that moreVal might be provided as -1 to avoid a requirement of all counted items having a specific more value. 🤔
```
int ResourceCount( realId, colour, moreVal, sectionId )
```

Reply:
Resolving these as is, as I don't want to require users to pass in the specific value (or some magic constant) in the arguments from js scripts